src/macho.rs

```diff
@@ -1,10 +1,23 @@
+use crate::error::{GetSymbolsError, Result};
 use compact_symbol_table::CompactSymbolTable;
 use object::{MachOFile, Object};
-
-pub fn get_compact_symbol_table(buffer: &[u8], breakpad_id: &str) -> Option<CompactSymbolTable> {
-    let macho_file = MachOFile::parse(buffer).ok()?;
-    if format!("{:X}0", macho_file.mach_uuid()?.simple()) != breakpad_id {
-        return None;
+pub fn get_compact_symbol_table(buffer: &[u8], breakpad_id: &str) -> Result<CompactSymbolTable> {
+    let macho_file = MachOFile::parse(buffer).or(Err(GetSymbolsError::InvalidInputError(
+        "Could not parse Mach-O header",
+    )))?;
+    let macho_id = match macho_file.mach_uuid() {
```

Comment:
Sorry for asking you to change this piece again, but there might actually be a better way to do this:

```rust
let macho_id = macho_file.mach_uuid().ok_or_else(|| GetSymbolsError::InvalidInputError("..."))?;
```